### PR TITLE
data API PoC

### PR DESCRIPTION
**Pull request by samurex**
Co-authored-by: elf Pavlik <elf-pavlik@hackers4peace.net>

### CODE REVIEW

**Comment by elf-pavlik** [comment]
I see two main advantages of this direction:

1) No need to make up some IriTemplates [example](https://github.com/janeirodigital/sai-impl-web/pull/15/files#diff-1bb1058564acdfb935d41de30e50c39ceedd63f443993cbd15806aeec94c2743L22)
2) Logic responsible for that API could be later moved from backend to frontend and even made that choice configurable per deployment

**Comment by elf-pavlik** [comment]
Corresponding backend draft PR https://github.com/janeirodigital/sai-impl-service/pull/42

File: src/app/services/data.service.ts [hunk 1]
```diff
@@ -1,24 +1,50 @@
...
+    const response = await this.solidClient.fetch(`${ENV.SRV_BASE}/api`, options)
+    return response.json()
+  }
 
-  getApplicationProfiles(): Observable<Application[]> {
-    return this.http.get<Application[]>('/api/applications', { responseType: 'json'});
+  async getApplicationProfiles(): Promise<Application[]> {
+    const options = {
+      ...commonOptions,
+      body: JSON.stringify({type: DataActions.applicationsPanelLoaded.type}),
```
Comment (angelaraya) [inline src/app/services/data.service.ts:32]: Do not use the `DataActions` type as it is an internal component of the UI and should not be exposed to the BE or any other component outside of the UI.

We discussed using `MessageType` instead. Another alternative is to just use an object with a const `type` property.

```js
body: JSON.stringify({type: GET_APPLICATIONS})
```
A third shared package where `GET_APPLICATIONS` is defined can be created and shared between any modules that use it to interact with one another.

File: src/app/services/data.service.ts [hunk 2]
```diff
@@ -1,24 +1,50 @@
...
   }
 
-  // TODO (angel) typing
-  getDescription(applicationId: string, lang: string): Observable<[]> {
-    return this.http.get<[]>(`/api/descriptions/${encodeURI(applicationId)}/${lang}`, {responseType: 'json'});
+  async getDescription(applicationId: string, lang: string): Promise<Description[]> {
+    const options = {
+      ...commonOptions,
+      body: JSON.stringify({
+        type: DescActions.descriptionsNeeded.type,
```
Comment (angelaraya) [inline src/app/services/data.service.ts:42]: Same as with the `applicationPanelLoaded` scenario above.

**Review by angelaraya — CHANGES_REQUESTED** [review]

**Comment by elf-pavlik** [comment]
The latest version uses a shared module from the corresponding service draft PR. For now, it relies on running `yarn link` or `npm link` manually.

**Pull request merged**